Source/Diagnostics/Diagnostics.cpp

```diff
@@ -43,6 +43,18 @@ Diagnostics::BaseReadParameters ()
         m_varnames = {"Ex", "Ey", "Ez", "Bx", "By", "Bz", "jx", "jy", "jz"};
     }
 
+    // default for writing field output is 1
+    int write_fields = 1;
+    pp.query("write_fields", write_fields);
+    if (write_fields == 0) {
+        if (m_format == "checkpoint") {
+            amrex::Abort("For checkpoint format, write_fields flag must be 1.");
+        }
+        // if user-defined value for write_fields == 0, then clear species vector
```

Comment:
```suggestion
        // if user-defined value for write_fields == 0, then clear field vector
```
I guess that `species vector` should be changed with something like this